services/motion/builtin/move_request.go

```diff
@@ -177,33 +181,158 @@ func (ms *builtIn) newMoveOnGlobeRequest(
 	} // Note: this is only for diff drive, not used for PTGs
 	ms.logger.Debugf("base limits: %v", limits)
 
+	replanCostFactor := defaultReplanCostFactor
+	if extra != nil {
+		if profile, ok := extra["motion_profile"]; ok {
+			motionProfile, ok := profile.(string)
+			if !ok {
+				return nil, errors.New("could not interpret motion_profile field as string")
+			}
+			kinematicsOptions.PositionOnlyMode = motionProfile == motionplan.PositionOnlyMotionProfile
+		}
+		if costFactorRaw, ok := extra["replan_cost_factor"]; ok {
```

Comment:
You can say no because its not a part of this PR but I noticed that the max replans parsing is done outside the context of creating the move on globe request.  Could you also move that into this function?

Reply:
Also why not just move this into the relativeMoveRequest from absolute?  I would like to see everything parsed out of the extra param in the same place

Reply:
Yeah that's a good call. Will do

Reply:
Nick's PR also does this